packages/@okta/vuepress-site/code/java/spring_security_saml/index.md

```diff
@@ -7,217 +7,385 @@ excerpt: >-
+
+   ```shell
+   openssl req -newkey rsa:2048 -nodes -keyout local.key -x509 -days 365 -out local.crt
+   ```
+
+1. Copy the generated files to your app's `src/main/resources` directory. Configure `signing` and `singlelogout` in `application.yml`:
+
+   ```yaml
+   spring:
+     security:
+       saml2:
+         relyingparty:
+           registration:
+             okta:
+               signing:
+                 credentials:
+                   - private-key-location: classpath:local.key
+                     certificate-location: classpath:local.crt
+               singlelogout:
+                 binding: POST
+                 response-url: "{baseUrl}/logout/saml2/slo"
+               assertingparty:
+                 metadata-uri: <your-metadata-uri>
+   ```
+
+1. Upload the `local.crt` to your Okta app. Select *Enable Single Logout* and use the following values:
+
+   * Single Logout URL: `\http://localhost:8080/logout/saml2/slo`
+   * SP Issuer: `\http://localhost:8080/saml2/service-provider-metadata/okta`
+
+1. Finish configuring your Okta app, restart your Spring Boot app, and the logout button should work.
+
+   <div class="three-quarter border">
+
+   ![Sign out success](/img/saml/spring-security-saml-logout-success.png)
+
+   </div>
 
-Before we can configure Spring Security SAML, we need to set up an application in Okta that connects to Spring Security SAML.
+### Customize authorities with Spring Security SAML
 
-In SAML terminology, what we are doing here is configuring Okta, which is our SAML Identity Provider (SAML IdP), with the details of Spring Security SAML that is the new SAML Service Provider (SAML SP) that you create next.
+When you sign in, the resulting page shows that you have a `ROLE_USER` authority. However, when you assigned users to the app, you gave access to `Everyone`. You can configure your SAML app on Okta to send a user's groups as an attribute, and add other attributes like name and email.
 
-Here is how to configure Okta:
+1. Edit your Okta app's SAML settings and fill in the **Group Attribute Statements** section.
 
-**1.** Sign in to your Okta organization as a user with administrative privileges. If you don't have an Okta organization, you can [create a free Okta Developer Edition organization](https://developer.okta.com/signup/).
+   * Name: `groups`
+   * Name format: `Unspecified`
+   * Filter: `Matches regex` and use `.*` for the value
 
-**2.** Click **Admin** to open the Admin Console. Go to **Applications** > **Applications**.
+1. Just above, you can add other attribute statements. For instance:
+   * Email:
+      * Name: `email`
+      * Name format: `Unspecified`
+      * Value: `user.email`
+   * First name:
+      * Name: `firstName`
+      * Name format: `Unspecified`
+      * Value: `user.firstName`
+   * Last name:
+      * Name: `lastName`
+      * Name format: `Unspecified`
+      * Value: `user.lastName`
```

Comment:
I created a table for these values in my blog post. Do you think that looks better than this? https://developer.okta.com/blog/2022/08/05/spring-boot-saml#customize-authorities-with-spring-security-saml

Reply:
Yup. Adding it into next commit